src/rinterface_extra.c

```diff
@@ -2380,6 +2380,12 @@ static int R_igraph_errors_count = 0;
 static char R_igraph_warning_reason[4096];
 static int R_igraph_warnings_count = 0;
 
+static int R_igraph_in_r_check = 0;
```

Comment:
Nitpick: since this seems to be affectively a boolean, why not use a C99 `bool` here? If you do, `#include <stdbool>`.